Adds use cases to website documentation (#137)
slsa-framework/slsa: docs/use-cases/consuming-third-party-software.md

---
layout: use-case
---

# Use cases

## Consuming third party software

A developer (e.g. FooCo) using third party software (e.g. HowDo developed by HowInc) wants to ensure it hasn’t been tampered with before using it. They could take these steps to protect the organisation from the software they consume without having to rely on any trust in intermediate package repositories.

| Subject   | Description                   |
|:----------|:------------------------------|
| **Users** | Software consumers            |
| **Goals** | Protecting against tampering  |

### How to do it

> -   Requesting HowInc publish the [in-toto SLSA Provenance](https://slsa.dev/provenance) and any additional attestations (such as [source control attestations-(https://github.com/in-toto/attestation/issues/47)) for HowDo each time it’s released

Reviewer: nit: the source control attestations link is broken.  I think you need to replace the - with [

> -   Requesting HowInc publish the [in-toto SLSA Provenance](https://slsa.dev/provenance) and any additional attestations (such as [source control attestations-(https://github.com/in-toto/attestation/issues/47)) for HowDo each time it’s released

Reviewer 2: ```suggestion
-   Requesting HowInc publish the [in-toto SLSA Provenance](https://slsa.dev/provenance) and any additional attestations (such as [source control attestations](https://github.com/in-toto/attestation/issues/47)) for HowDo each time it’s released
```

-   Requesting HowInc publish the public keys its builder uses to sign the attestations
-   Requesting HowInc confirm what SLSA level their builder and source control system meet
-   Determining what policy to apply to HowDo
    -   This policy could be created on first use based on the data in the in-toto SLSA Provenance. Any significant deviations (e.g. the builder or source repository changed) would cause failure
    -   Or HowInc could publish a suggested policy for HowDo users on their website
-   Establishing a secure control-point that any HowDo versions/artifacts must pass through to be used

>     -   e.g. on import to a local Docker registory

Reviewer 2: ```suggestion
    -   e.g. on import to a local Docker registry
```

> -   Having the control-point check the candidate HowDo against its provenance, checking it against the policy mentioned above

Reviewer 2: ```suggestion
-   Having the control-point check the candidate HowDo and its provenance (as well as any other published attestations) against the policy mentioned above, ensuring any attestations evaluated by the policy are signed by HowInc's public key(s)
```
Expanded to:
a) indicate that artifacts and their provenance (as well as other attestations) are evaluated
b) remind folks to verify signed data

---
layout: use-case
---

# Use cases

## Consuming third party software

> A software development organisation developer (e.g. FooCo) using third party software (e.g. HowDo developed by HowInc) wants to ensure it hasn’t been tampered with before using it. They could take the following steps to check, protecting the organisation from the software they consume without having to rely on any trust in intermediate package repositories.

Reviewer 3: 1. This also relates to the comment below about "organisation" <> "developer". We're going to switch to "developer" so that both individuals and individuals-in-organisations feel included. 
2. Will change "They could take the following steps to check, protecting the organisation from the software.." to "They could take these next steps to protect the organisation from the software..." to improve the second sentence

| Subject   | Description                   |
|:----------|:------------------------------|
| **Users** | Software consumers            |
| **Goals** | Protecting against tampering  |

### How to do it

-   Requesting HowInc publish the [in-toto SLSA Provenance] and any additional attestations (e.g. source control attestations) for HowDo each time it’s released
-   Requesting HowInc publish the public keys its builder uses to sign the attestations

>     -   Determine how to convey these keys

Reviewer 3: We're trying to keep the main text of the use cases without open action items as we believe that will work for more users. That can mean putting the open action items into the limitations section, as they're important to know. I'll move the <determine how to convey these keys> bit to the limitation section and link to the related issue: https://github.com/slsa-framework/slsa/issues/101

-   Requesting HowInc confirm what SLSA level their builder and source control system meet
-   Determining what policy to apply to HowDo
    -   This policy could be created on first use based on the data in the [in-toto SLSA Provenance]. Any significant deviations (e.g. the builder or source repository changed) would cause failure
    -   Or HowInc could publish a suggested policy for HowDo users on their website

> -   Establishing a secure control-point that any HowDo users must pass through to be used

Reviewer 3: will change  "Establishing a secure control-point that any HowDo users must pass through to be used" to "Establishing a secure control-point that any HowDo versions/artifacts must pass through to be used"

    -   e.g. on import to a local Docker registory

> -   Having the control-point check the candidate HowDo against its provenance, checking it -   against the policy mentioned above

Reviewer 3: Another typo that we've let snuck in, sorry, will delete

> -   Only importing the siftware artifact if all the above checks pass

Reviewer 3: 1. agh, typo. Sorry!
2. Checking back to the original the intent here is everything in the bulleted list so reading it that way is what we want
